engine/access/state_stream/backend.go

```diff
@@ -154,9 +175,12 @@ func (b *StateStreamBackend) getStartHeight(startBlockID flow.Identifier, startH
 		return header.Height, nil
 	}
 
-	// next, if the start height is provided, use that
-	// heights that are in the future or before the root block will result in an error
+	// heights that have not been indexed yet will result in an error
 	if startHeight > 0 {
+		if startHeight < b.rootBlockHeight {
+			return 0, status.Errorf(codes.InvalidArgument, "start height must be greater than or equal to the spork root height %d", b.rootBlockHeight)
```

Comment:
maybe log the start height in the message as well?

Reply:
the start height is part of the request, so the client already knows it.